lms/resources/_js_config.py

```diff
@@ -16,17 +16,20 @@ class JSConfig:  # pylint:disable=too-many-instance-attributes
     def __init__(self, context, request):
         self._context = context
         self._request = request
-        self._lti_user = request.lti_user
-        self._h_user = request.lti_user.h_user
-        self._consumer_key = (
-            request.lti_user.oauth_consumer_key if request.lti_user else None
-        )
         self._authority = request.registry.settings["h_authority"]
-
         self._grading_info_service = request.find_service(name="grading_info")
-        self._ai_getter = request.find_service(name="ai_getter")
         self._h_api = request.find_service(name="h_api")
 
+        self._lti_user = request.lti_user
+        if self._lti_user:
+            self._h_user = self._lti_user.h_user
+            self._consumer_key = self._lti_user.oauth_consumer_key
+            self._ai_getter = request.find_service(name="ai_getter")
+        else:
+            self._h_user = None
+            self._consumer_key = None
+            self._ai_getter = None
```

Comment:
I'd like to avoid if/else login in `__init__` methods.

What's the reason for this change?

I'm guessing that, for authorization error dialog requests, `request.find_service(name="ai_getter")` crashes? And `request.lti_user` would be `None` and `request.lti_user.h_user` would crash.

Could we instead:

* Leave `self._lti_user = request.lti_user` alone. I think this one is fine? `self._lti_user` will just be `None` for auth error dialog requests

* Move `request.find_service(name="ai_getter")` and `request.lti_user.h_user` and `self._lti_user.oauth_consumer_key` into properties:

  ```python
  @property
  def _ai_getter(self):
      return self._request.find_service(name="ai_getter")

  @property
  def _h_user(self):
      return self._lti_user.h_user

  ...
  ```

  You could add docstrings explaining that these properties should only be accessed during authenticated requests.

  Accessing the properties during an auth error dialog request would crash. But the auth error dialog view presumably does not access these properties.

  It's not necessary to implement caching for these properties. The efficiency saving would be negligible. pyramid_services already caches services.